packages/paste-core/components/button/src/types.ts

```diff
@@ -34,32 +34,30 @@ export interface ButtonContentsProps {
 }
 
 export interface DirectButtonProps extends HTMLPasteProps<'button'>, Pick<BoxProps, 'element'> {
-  size: ButtonSizes;
-  children: React.ReactNode;
   as?: keyof JSX.IntrinsicElements;
+  buttonState: ButtonStates;
+  children: React.ReactNode;
+  disabled?: boolean;
   fullWidth?: boolean;
   href?: string;
+  pressed?: boolean;
+  size: ButtonSizes;
   tabIndex?: ButtonTabIndexes;
-  disabled?: boolean;
+  target?: string;
   type?: ButtonTypes;
-  buttonState: ButtonStates;
   variant: ButtonVariants;
-  pressed?: boolean;
-  target?: string;
 }
 
-export interface ButtonProps extends HTMLPasteProps<'button'>, Pick<BoxProps, 'element'> {
-  variant: ButtonVariants;
-  size?: ButtonSizes;
-  as?: keyof JSX.IntrinsicElements;
-  fullWidth?: boolean;
-  href?: string;
-  tabIndex?: ButtonTabIndexes;
-  disabled?: boolean;
-  type?: ButtonTypes;
-  loading?: boolean;
-  children: React.ReactNode;
-  i18nExternalLinkLabel?: string;
-  pressed?: boolean;
-  target?: string;
-}
+type BaseVariantsButtonProps = {
+  variant?: Omit<ButtonVariants, 'reset'>;
+};
+type ResetVariantButtonProps = BoxStyleProps & {
+  variant?: 'reset';
+};
+
+export type ButtonProps = (BaseVariantsButtonProps | ResetVariantButtonProps) &
```

Comment:
Do we want to add docs for this? An example of using reset button? Or keep it under wraps for now?